### ISSUE TITLE
Koreader crashes when asked to update itself on nightly to another nightly [Kindle Scribe]

### ISSUE THREAD

**Issue by arooni** [koreader/koreader#11240]
* KOReader version: 2023.10-55
* Device: Kindle Scribe

#### Issue
Pick nightly updates; not stable.

#### Steps to reproduce
ask to update
app crashes and kicked out of KUAL & koreader

##### `crash.log` (if applicable)
`crash.log` is a file that is automatically created when KOReader crashes. It can normally be found in the KOReader directory:

* `/mnt/private/koreader` for Cervantes
* `koreader/` directory for Kindle
* `.adds/koreader/` directory for Kobo
* `applications/koreader/` directory for Pocketbook

Android logs are kept in memory. Please go to [Menu] → Help → Bug Report to save these logs to a file.

Please try to include the relevant sections in your issue description.
You can upload the whole `crash.log` file (zipped if necessary) on GitHub by dragging and dropping it onto this textbox.

If your issue doesn't directly concern a Lua crash, we'll quite likely need you to reproduce the issue with *verbose* debug logging enabled before providing the logs to us.
To do so, from the file manager, go to [Tools] → More tools → Developer options, and tick both `Enable debug logging` and `Enable verbose debug logging`.
You'll need to restart KOReader after toggling these on.

If you instead opt to inline it, please do so behind a spoiler tag:
<details>
  <summary>crash.log</summary>

```
<Paste crash.log content here>


             12/17/23-21:47:28 DEBUG setDirty via a func from widget table: 0xb5806fb0
12/17/23-21:47:28 DEBUG painting widget: table: 0xb5806fb0
12/17/23-21:47:28 DEBUG _refresh: Enqueued ui update for region 289 1167 1282 145 dithering: false
12/17/23-21:47:28 DEBUG refresh: ui-mode 289 1167 1282 145 nil
12/17/23-21:47:28 DEBUG refresh: wait for submission of (previous) marker 30
12/17/23-21:47:28 DEBUG mxc_update: 1282x145 region @ (289, 1167) with marker 31 (WFM: 257 & UPD: 0)
12/17/23-21:47:28 DEBUG close widget: table: 0xb5806fb0
12/17/23-21:47:28 DEBUG setDirty via a func from widget nil
12/17/23-21:47:28 DEBUG Lower widget ReaderUI covers the full screen
12/17/23-21:47:28 DEBUG setDirty nil from widget ReaderUI w/ NO region; dithering: nil
12/17/23-21:47:28 DEBUG setDirty nil from widget table: 0xab2f0c48 w/ NO region; dithering: nil
./luajit: frontend/ui/otamanager.lua:161: loop or previous error loading module 'socketutil'
stack traceback:
        [C]: in function 'require'
        frontend/ui/otamanager.lua:161: in function 'checkUpdate'
        frontend/ui/otamanager.lua:226: in function 'fetchAndProcessUpdate'
        frontend/ui/otamanager.lua:487: in function 'callback'
        frontend/ui/network/manager.lua:593: in function 'runWhenOnline'
        frontend/ui/otamanager.lua:491: in function 'callback'
        frontend/ui/widget/touchmenu.lua:921: in function 'onMenuSelect'
        frontend/ui/widget/touchmenu.lua:216: in function 'handleEvent'
        frontend/ui/widget/container/inputcontainer.lua:262: in function 'handleEvent'
        frontend/ui/widget/container/widgetcontainer.lua:82: in function 'propagateEvent'
        frontend/ui/widget/container/widgetcontainer.lua:100: in function 'handleEvent'
        ...
        frontend/ui/widget/container/widgetcontainer.lua:100: in function 'handleEvent'
        frontend/ui/widget/container/widgetcontainer.lua:82: in function 'propagateEvent'
        frontend/ui/widget/container/widgetcontainer.lua:100: in function 'handleEvent'
        frontend/ui/uimanager.lua:907: in function 'sendEvent'
        frontend/ui/uimanager.lua:53: in function '__default__'
        frontend/ui/uimanager.lua:1436: in function 'handleInputEvent'
        frontend/ui/uimanager.lua:1534: in function 'handleInput'
        frontend/ui/uimanager.lua:1578: in function 'run'
        ./reader.lua:265: in main chunk
        [C]: at 0x00013ebd
lipc-wait-event exited normally with status: 0
[root@kindle koreader]# Vk



```
</details>

**Comment by arooni** [comment]
Also seems to crash when asking it to keep progress in sync:

```
12/17/23-22:14:03 DEBUG Explicit widgetInvert: table: 0xaba48de0 @ 16 482
12/17/23-22:14:03 DEBUG _refresh: Enqueued fast update for region 16 482 1828 140 dithering: false
12/17/23-22:14:03 DEBUG setDirty fast from widget nil w/ region 16 482 1828 140 dithering: nil
12/17/23-22:14:03 DEBUG refresh: fast 16 482 1828 140 nil
12/17/23-22:14:03 DEBUG mxc_update: 1828x140 region @ (16, 482) with marker 9 (WFM: 1 & UPD: 0)
./luajit: plugins/kosync.koplugin/KOSyncClient.lua:3: loop or previous error loading module 'socketutil'
stack traceback:
        [C]: in function 'require'
        plugins/kosync.koplugin/KOSyncClient.lua:3: in main chunk
        [C]: in function 'require'
        plugins/kosync.koplugin/main.lua:718: in function 'getProgress'
        plugins/kosync.koplugin/main.lua:240: in function 'callback'
        frontend/ui/widget/touchmenu.lua:921: in function 'onMenuSelect'
        frontend/ui/widget/touchmenu.lua:216: in function 'handleEvent'
        frontend/ui/widget/container/inputcontainer.lua:262: in function 'handleEvent'
        frontend/ui/widget/container/widgetcontainer.lua:82: in function 'propagateEvent'
        frontend/ui/widget/container/widgetcontainer.lua:100: in function 'handleEvent'
        frontend/ui/widget/container/widgetcontainer.lua:82: in function 'propagateEvent'
        ...
        frontend/ui/widget/container/widgetcontainer.lua:100: in function 'handleEvent'
        frontend/ui/widget/container/widgetcontainer.lua:82: in function 'propagateEvent'
        frontend/ui/widget/container/widgetcontainer.lua:100: in function 'handleEvent'
        frontend/ui/uimanager.lua:907: in function 'sendEvent'
        frontend/ui/uimanager.lua:53: in function '__default__'
        frontend/ui/uimanager.lua:1436: in function 'handleInputEvent'
        frontend/ui/uimanager.lua:1534: in function 'handleInput'
        frontend/ui/uimanager.lua:1578: in function 'run'
        ./reader.lua:265: in main chunk
        [C]: at 0x00013ebd
lipc-wait-event exited normally with status: 0
[root@kindle koreader]#

```

**Comment by Frenzie** [comment]
That's strange, socketutil hasn't changed in months and luasocket a bit more recently perhaps but if so it should have issues everywhere.

Could you try to overwrite the files manually to see what happens? It kind of sounds to me like the files may have become corrupted somehow; @NiLuJe might have something more insightful to say about that.

http://build.koreader.rocks/download/nightly/v2023.10-64-g48197805a_2023-12-17/

**Comment by arooni** [comment]
> That's strange, socketutil hasn't changed in months and luasocket a bit more recently perhaps but if so it should have issues everywhere.
> 
> Could you try to overwrite the files manually to see what happens? It kind of sounds to me like the files may have become corrupted somehow; @NiLuJe might have something more insightful to say about that.
> 
> http://build.koreader.rocks/download/nightly/v2023.10-64-g48197805a_2023-12-17/

Yes I think this is (yet) another case of "arooni user error".  I extracted the archive directly to the Kindle; verified it had no issue making network calls; then copied over the settings and it works great!  Can check updates without issue.  Can do progress sync without issue.  Can even export to Readwise!  

I haven't used koreader in awhile since I picked up the Scribe as I thought jailbreak would be impossible.  Luckily I got a refurbished device recently that was vulnerable to LanguageBreak and spent the weekend getting that done There were definitely a few sketchy moments getting to this state -- but so worth it.  So happy now!  I missed KoReader more than I thought, I love the customizations and even simple things like the status bar are so much better than the stock Kindle experience.

This is so much easier now that I have ssh access over wifi to the kindle; sorry Amazon; I'm doing some tinkering  with your device outside your walled garden.  

Set up syncthing to move my books around all devices much easier.  ebook heaven !  :)
